lib/core/src/Cardano/Wallet/DB/Sqlite.hs

```diff
@@ -0,0 +1,125 @@
+{-# LANGUAGE DataKinds #-}
+{-# LANGUAGE DeriveGeneric #-}
+{-# LANGUAGE FlexibleInstances #-}
+{-# LANGUAGE GADTs #-}
+{-# LANGUAGE GeneralizedNewtypeDeriving #-}
+{-# LANGUAGE MultiParamTypeClasses #-}
+{-# LANGUAGE QuasiQuotes #-}
+{-# LANGUAGE TemplateHaskell #-}
+{-# LANGUAGE TypeApplications #-}
+{-# LANGUAGE TypeFamilies #-}
+
+module Cardano.Wallet.DB.Sqlite where
+
+import Prelude
+
+import Cardano.Wallet.DB.SqliteTypes
+    ( AddressScheme, TxId )
+import Data.Text
+    ( Text )
+import Data.Time.Clock
+    ( UTCTime )
+import Data.Word
+    ( Word32, Word64 )
+import Database.Persist.TH
+import GHC.Generics
+    ( Generic (..) )
+
+import qualified Cardano.Wallet.Primitive.Types as W
+
+-- fixme: need tables for wallet AddressPool
+
+share
+    [ mkPersist sqlSettings { mpsPrefixFields = False }
+    , mkMigrate "migrateAll"
+    ]
+    [persistLowerCase|
+
+Wallet
+  walId                W.WalletId              sql=wallet_id
+  walName              Text                    sql=name
+  walPassphraseLastUpdatedAt   UTCTime         sql=passphrase_last_updated_at
+  walStatus            W.WalletState           sql=status
+  walDelegation        Text Maybe              sql=delegation
+  walAddressScheme     AddressScheme           sql=address_discovery
+
+  Primary walId
+  deriving Show Generic
+
+WalletPrivateKey                               sql=private_key
+  walPrivateKeyWalId   W.WalletId              sql=wallet_id
+  walPrivateKey        Text                    sql=private_key
+
+  Primary walPrivateKeyWalId
+  Foreign Wallet fk_wallet_private_key walPrivateKeyWalId
+
+  deriving Show Generic
+
+TxMeta
+  txId                  TxId                   sql=tx_id
+  txMetaWalletId        W.WalletId             sql=wallet_id
+  txStatus              W.TxStatus             sql=status
+  txMetaDirection       W.Direction            sql=direction
+  txMetaSlotId          W.SlotId               sql=slot_id
+  txMetaAmount          Word64                 sql=amount
+
+  Primary txId txMetaWalletId
+  Foreign Wallet fk_wallet_tx_meta txMetaWalletId
+  deriving Show Generic
+
+TxInput
+  txInputTxId           TxId                   sql=tx_id
+  txInputSourceTxId     TxId                   sql=source_id
+  txInputSourceIndex    Word32                 sql=source_index
+  txInputAddress        Text                   sql=address
+  txInputAmount         Word64                 sql=amount
+
+  Primary txInputTxId txInputSourceTxId txInputSourceIndex
+  -- constraint: tx_id must exist in TxMeta or PendingTx
```

Comment:
@rvl @jonathanknowles are we sure we need this constraint? I made it required in utxo

Reply:
I removed the comment.